src/components/calcite-panel/calcite-panel.tsx

```diff
@@ -0,0 +1,186 @@
+import { Component, Element, Event, EventEmitter, Host, Prop, h } from "@stencil/core";
+import { CSS, TEXT } from "./resources";
+import { getElementDir } from "../utils/dom";
+import classnames from "classnames";
+import { CSS_UTILITY } from "../utils/resources";
+import { VNode } from "@stencil/core/dist/declarations";
+import { CalciteTheme } from "../interfaces";
+import CalciteIcon from "../utils/CalciteIcon";
+import { x16 } from "@esri/calcite-ui-icons";
+
+const SLOTS = {
+  headerContent: "header-content",
+  headerLeadingContent: "header-leading-content",
+  headerTrailingContent: "header-trailing-content",
+  footer: "footer"
+};
+
+@Component({
+  tag: "calcite-panel",
+  styleUrl: "calcite-panel.scss",
+  shadow: true
+})
+export class CalcitePanel {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * Displays a close button in the trailing side of the header.
+   */
+  @Prop({ reflect: true }) dismissible = false;
+
+  /**
+   * 'Close' text string for the close button.
+   */
+  @Prop() textClose = TEXT.close;
+
+  /**
+   * Used to set the component's color scheme.
+   */
+  @Prop({ reflect: true }) theme: CalciteTheme;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLCalcitePanelElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * Emitted when the close button has been clicked.
+   */
+
+  @Event() calcitePanelDismiss: EventEmitter;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  dismiss = (): void => {
+    this.calcitePanelDismiss.emit();
```

Comment:
boulder: imo, this should toggle some internal State that hides the panel automatically. I wouldn't want to have to wire up an event handler to close it once I made it dismissible.